### DISCUSSION TITLE
secli - Bash CLI to manage SoftEther VPN Server using JSON-RPC

### DISCUSSION THREAD

**Discussion by shakibamoshiri** [SoftEtherVPN/SoftEtherVPN#1786] [General]
As full time Terminal user I would like to manage SE servers from a Terminal even SE server manger is awesome and makes things much simpler.
The idea began when I needed to disable myself automatically, if my traffic usage exceeded a certain number (e.g. 10 Gig)
There was no such feature and I had two options to go with
1. writing a or some bash scripts as wrapper round "vpncmd"
2. working with RPC-JSON API

The #1 is simple and could be done in few days or weeks, but #2 was harder and more challenging , so I toke #2.
And here is the result
https://github.com/shakibamoshiri/secli

At the moment mostly I am looking for
- bug reports to fix
- common expect functionalities to see and implement (if possible)

I think SoftEther VPN a great tool/technology and deserves having a CLI to be managed with.
So please share your thoughts and ideas to make secli more reliable

Regards

**Comment by shakibamoshiri** [comment]
I know nowadays  most modern applications either they have  
- a web application dashboard, or 
- a working CLI

My experience/knowledge with CLI and Bash was much better than ReactJS/Frontend. if I knew more ReactJS, I would go with ReactJS.

Thus I think core developers of SE server should focus on  
- SE server
- SE server API

and not other things, If we have these two solid and reliable, the others can make  
- CLIs
- Web Apps
- Win Apps

As an example for **WireGuard** I saw people requesting you to add it to Win GUI. It takes time from core developers to simultaneously work on Both server and server manger for Windows. while it toke me a few hours using "vpncmd" to have a bash CLI to automatically generate WG configuration.
If SE server had API for WG, I could do it it less time.

**Comment by metalefty** [comment]
Nice work, I'll try.
